### ISSUE TITLE
Sometimes Singular creates unknown coefficient fields, which make Oscar unhappy

### ISSUE THREAD

**Issue by thofma** [oscar-system/Oscar.jl#3088]
Best illustrated with an example:
```
julia> Qx, x = QQ["x"]; K, = number_field([x - 1, x^2 + 2], cached = false); # or any complicated field that translates into a generic `N_Field`

julia> R, = polynomial_ring(K, 2);

julia> I = ideal(R, [gen(R, 1)]);

julia> S, = quo(R, I);

julia> Oscar.singular_poly_ring(R) |> coefficient_ring |> typeof
Singular.N_Field{NfAbsNSElem}

julia> Oscar.singular_poly_ring(S) |> coefficient_ring |> typeof
Singular.N_UnknownSingularCoefficientRing
```

This happens, because for the construction of the Singular quotient ring we go via:
https://github.com/oscar-system/Oscar.jl/blob/32b7e894489f03dc7e90a60d9e8944bb3929839c/src/Rings/MPolyQuo.jl#L31
(this is the Singular quotient ring, of which the `coefficient_ring` is of the `N_Unknown[...]` type.

As a consequence, on the Singular side we don't have the identical coefficient ring, which yields problems like #3080.

This is ultimately a Singular.jl issue, but I guess here the issue gets more attention. I think the discussion in https://github.com/oscar-system/Singular.jl/issues/187#issuecomment-617095888 and https://github.com/oscar-system/Singular.jl/pull/336 is relevant.

One option is ditch the generic `N_Field` interface and just refuse to do anything with coefficient rings that cannot be handled by native Singular coefficient fields. In most cases we can transform the input to the appropriate type. For the rest we will throw a "sorry, can't do this" error. This is a rather drastic solution, but maybe the only realistic one given our manpower?

Any thoughts @fieker @fingolfin @hannes14?

**Comment by thofma** [comment]
Here is another simple example. Lots of things are broken for prime fields of large characteristic:
```
julia> F = GF(next_prime(ZZ(2)^100))
Finite field of characteristic 1267650600228229401496703205653

julia> Fx, (x, y) = F["x", "y"];

julia> I = ideal(Fx, [x, y]);

julia> Q, = quo(Fx, I);

julia> J = ideal(Q, [Q(x)]);

julia> Q(x) in J
ERROR: MethodError: no method matching (::Singular.N_UnknownSingularCoefficientRing)(::FpFieldElem)
```

**Comment by HechtiDerLachs** [comment]
Is there any progress on this issue. The workarounds in #3173 only cause other troubles and it seems that I will never get the tests to run there.

**Comment by fieker** [comment]
it should be, as of last Wedesday(??) be fixed. We confirm and close.
Thanks @hannes14

**Comment by fieker** [comment]
update: still open(ish), but making progress

**Comment by hannes14** [comment]
Status: fixed with master branch of Oscar, not fixed with released version of Oscar

**Comment by lgoettgens** [comment]
I can confirm the fix on e86fe099fce2a0b6f5d92b121178d3a6ab625fd4.
